src/lib/statusandheaders.ts

```diff
@@ -202,6 +205,16 @@ function splitRemainder(str: string, sep: string, limit: number) {
   return newParts;
 }
 
+// ===========================================================================
+function encodeLatin1(value: string) {
+  const buf = new TextEncoder().encode(value);
+
+  let str = "";
+  buf.forEach((x) => (str += String.fromCharCode(x)));
+  //buf.forEach(x => str += x < 128 ? String.fromCharCode(x) : `\\x${x}`);
```

Comment:
I wonder if using [buffer.transcode()](https://nodejs.org/api/buffer.html#buffer_buffer_transcode_source_fromenc_toenc) might be a good idea here, as there might be edge cases where `TextEncoder().encode(value)` might encode certain characters as two bytes?

Reply:
Hmm, yeah, though buffer is not available in the browser, so would have to use the equivalent.

Reply:
Ah true, so used to working with node I forgot! It is also an edge case within an edge case, so I trust how much attention you want to give this

Reply:
I guess if we change this, it will break this assertion from the test case:

```
function decodeLatin1(buf) {
  let str = "";
  for (let i = 0; i < buf.length; i++) {
    str += String.fromCharCode(buf[i]);
  }
  return str;
}
d = decodeLatin1(new TextEncoder().encode("https://wiki.archlinux.jp/index.php/Arch_User_Repository?rdfrom=https%3A%2F%2Fwiki.archlinux.org%2Findex.php%3Ftitle%3DAUR_Metadata_%28%25E6%2597%25A5%25E6%259C%25AC%25E8%25AA%259E%29%26redirect%3Dno#AUR_メタデータ"));

// this matches the headers from node fetch:
a = await fetch("https://wiki.archlinux.org/title/AUR_Metadata_(%E6%97%A5%E6%9C%AC%E8%AA%9E)", {redirect: "manual"})
assert(a.headers.get("location") === d)
```

If we do: `decodeLatin1(Buffer.from("https://wiki.archlinux.jp/index.php/Arch_User_Repository?rdfrom=https%3A%2F%2Fwiki.archlinux.org%2Findex.php%3Ftitle%3DAUR_Metadata_%28%25E6%2597%25A5%25E6%259C%25AC%25E8%25AA%259E%29%26redirect%3Dno#AUR_メタデータ", "ascii"));` it will give a different result than what node fetch is doing..

Reply:
Resolved this in the most correct way, I think. Still encoding utf8-latin1 to store in Headers object, but then reencoding back when serializing